evaluate.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+
+import torch
+from torch.utils.data import Dataset, DataLoader, RandomSampler, SequentialSampler
+from main import NegCueDataset, BertForNegationCueClassification, MODEL_NAME
+from feature_embeddings import BertPrep
+from tqdm import tqdm
+from sklearn.metrics import classification_report
+
+if __name__ == "__main__":    
+    checkpoint_name = f"{MODEL_NAME}_{999}_{999}"
+    
+    lexicals = []
+    # lexicals = ["POS", "Possible_Prefix", "Possible_Suffix"]
+
+    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+
+    train_prep = BertPrep("data/SEM-2012-SharedTask-CD-SCO-training-simple-v2-features.csv", lexicals)
+    n_lexicals = 0 if len(lexicals) == 0 else sum(len(next(iter(v.values()))) for v in train_prep.feature_labels.values())
+    train_dataset = NegCueDataset(train_prep.preprocess_dataset(), n_lexicals=n_lexicals)
+
+    dev_prep = BertPrep("data/SEM-2012-SharedTask-CD-SCO-dev-simple-v2-features.csv", lexicals)
+    dev_dataset = NegCueDataset(dev_prep.preprocess_dataset(), n_lexicals=n_lexicals)
+    
+    dev_data_loader = DataLoader(dev_dataset, batch_size=1, num_workers=1)
+    
+    print(f"Loading checkpoint: {checkpoint_name}")
+    model = BertForNegationCueClassification.from_pretrained(
+        checkpoint_name,
+        num_labels=len(train_prep.tag2idx),
+        n_lexicals=n_lexicals
+    )
+    
+    model.to(device)
+
+    true_tags = []
+    pred_tags = []
+    
+    pred_tags_parsed = []
+    true_tags_parsed = []
+    VERBOSE=False
+
+    tokenizer = train_prep.tokenizer
+    inv_tag_enc = {v:k for k,v in train_prep.tag2idx.items()}
+    with torch.no_grad():
```

Comment:
I am not sure as well, but if optimizer is active it may count this as steps, which can mess it up. (Because for e.g. Adam optimizer keeps gradient from couple of steps before). But it is just my guess..